OpenRA.Mods.Common/Scripting/Global/MediaGlobal.cs

```diff
@@ -188,14 +188,24 @@ public bool PlayMovieInRadar(string movie, LuaFunction playComplete = null)
 			return true;
 		}
 
-		[Desc("Display a text message to the player.")]
-		public void DisplayMessage(string text, string prefix = "Mission", HSLColor? color = null)
+		[Desc("Display a text message to all players.")]
+		public void DisplayMessage(string text, string prefix = "Mission", HSLColor? color = null, int team = 0)
 		{
 			if (string.IsNullOrEmpty(text))
 				return;
 
 			Color c = color.HasValue ? HSLColor.RGBFromHSL(color.Value.H / 255f, color.Value.S / 255f, color.Value.L / 255f) : Color.White;
-			Game.AddChatLine(c, prefix, text);
+			var world = Context.World;
+
+			if (team == 0 || world.LobbyInfo.ClientWithIndex(world.LocalPlayer.ClientIndex).Team == team)
```

Comment:
`world.LocalPlayer` will be null during replays. If you used `RenderPlayer`, those messages could be seen in replays as well (with the correct player view selected). You'd still need a null check, though.